lib/src/consts.rs

```diff
@@ -122,16 +125,54 @@ lazy_static! {
     };
 }
 
-pub fn get_network_spec(network: Network) -> ChainSpec {
-    match network {
-        Network::Ethereum => ETH_MAINNET_CHAIN_SPEC.clone(),
-        Network::Holesky => ETH_HOLESKY_CHAIN_SPEC.clone(),
-        Network::TaikoA7 => TAIKO_A7_CHAIN_SPEC.clone(),
+#[derive(Clone, Debug)]
+pub struct SupportedChainSpecs(HashMap<String, ChainSpec>);
+
+impl SupportedChainSpecs {
+    pub fn default() -> Self {
+        SupportedChainSpecs(
+            vec![
```

Comment:
Small thing here, I would remove the `vec!` and just use a basic array as the vector is never used and a basic array can be iterated over just the same